### PR TITLE
Add google/fuzztest@20240813.0

### LINKED ISSUE #1391: wanted: google/fuzztest
### Module location

https://github.com/google/fuzztest

### Link to bzlmod issue in the module's repository

https://github.com/google/fuzztest/issues/950

### Any other context to provide?

_No response_

### Fund our work

- [ ] Sponsor our community's open source work by donating a [feature bounty](https://opencollective.com/bazel-rules-authors-sig/projects/bazel-central-registry)

### PR DESCRIPTION
**Pull request by udaya2899**
FIxes #1391

### CODE REVIEW

**Comment by bazel-io** [comment]
Hello @bazelbuild/bcr-maintainers, modules without existing maintainers (fuzztest) have been updated in this PR. Please review the changes.

File: modules/fuzztest/metadata.json [hunk 1]
```diff
@@ -0,0 +1,17 @@
+{
+    "homepage": "https://github.com/google/fuzztest",
+    "maintainers": [
+        {
+            "email": "lszekeres@google.com",
+            "github": "lszekeres",
```
Comment (udaya2899) [inline modules/fuzztest/metadata.json:6]: @lszekeres - Is it okay that I added you as a maintainer of this module?

**Comment by meteorcloudy** [comment]
It's listed as a dev_dependency, which won't be available if you are building fuzztest as a dependency. Maybe try build specific targets to avoid building the tests if it's not needed? Or you can use a test module setup to build the module as a root module.

File: modules/fuzztest/metadata.json (hunk 1 above)
Comment (lszekeres) [reply modules/fuzztest/metadata.json:6]: Yes, thanks!

File: modules/fuzztest/20240813.0/MODULE.bazel [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+bazel_dep(
+    name = "rules_proto",
+    version = "5.3.0-21.7",
+)
+
+bazel_dep(
+    name = "riegeli",
+    version = "0.0.0-20240606-973b6f0",
+    dev_dependency = True,
```
Comment (mering) [inline modules/fuzztest/20240813.0/MODULE.bazel:54]: This seems to be a required dependency for the two test targets as well.

Comment (udaya2899) [reply modules/fuzztest/20240813.0/MODULE.bazel:54]: Turned off dev_dependency

File: modules/fuzztest/20240813.0/patches/update_module_bazel_version.patch [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+ 
+ bazel_dep(
+
+@@ -51,7 +51,6 @@
+ bazel_dep(
+     name = "riegeli",
+     version = "0.0.0-20240606-973b6f0",
+-    dev_dependency = True,
+     repo_name = "com_google_riegeli",
+ )
```
Comment (mering) [inline modules/fuzztest/20240813.0/patches/update_module_bazel_version.patch]: This patch seems wrong. You probably want to add an unchanged context empty line (single space) to the end.

File: modules/fuzztest/20240813.0/patches/update_module_bazel_version.patch [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+ 
+@@ -46,7 +46,6 @@ bazel_dep(
+ bazel_dep(
+     name = "rules_proto",
+     version = "5.3.0-21.7",
+-    dev_dependency = True,
+ )
+ 
+ bazel_dep(
+
```
Comment (mering) [inline modules/fuzztest/20240813.0/patches/update_module_bazel_version.patch:19]: This line probably shouldn't be there.
```suggestion
```

File: modules/fuzztest/20240813.0/patches/update_module_bazel_version.patch (hunk 3 above)
Comment (udaya2899) [reply modules/fuzztest/20240813.0/patches/update_module_bazel_version.patch]: Done.

File: modules/fuzztest/20240813.0/patches/update_module_bazel_version.patch (hunk 4 above)
Comment (udaya2899) [reply modules/fuzztest/20240813.0/patches/update_module_bazel_version.patch]: AH! This was the missing bug. Thanks!

File: modules/fuzztest/20240813.0/patches/update_module_bazel_version.patch [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+ 
+ bazel_dep(
+@@ -51,7 +51,6 @@
+ bazel_dep(
+     name = "riegeli",
+     version = "0.0.0-20240606-973b6f0",
+-    dev_dependency = True,
+     repo_name = "com_google_riegeli",
+ )
+
```
Comment (mering) [inline modules/fuzztest/20240813.0/patches/update_module_bazel_version.patch:26]: This needs to contain a space. You might have to teach your editor to keep lines only consisting of a space in `.patch` files.

File: modules/fuzztest/20240813.0/patches/update_module_bazel_version.patch [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+     repo_name = "com_google_fuzztest",
+ )
+ 
+@@ -46,7 +46,6 @@ bazel_dep(
+ bazel_dep(
+     name = "rules_proto",
+     version = "5.3.0-21.7",
+-    dev_dependency = True,
+ )
+
```
Comment (mering) [inline modules/fuzztest/20240813.0/patches/update_module_bazel_version.patch:17]: Same here.

File: modules/fuzztest/20240813.0/patches/update_module_bazel_version.patch [hunk 7]
```diff
@@ -0,0 +1,26 @@
+--- MODULE.bazel
++++ MODULE.bazel
+@@ -1,6 +1,6 @@
+ module(
+     name = "fuzztest",
+-    version = "head",
++    version = "20240813.0",
+     repo_name = "com_google_fuzztest",
+ )
+
```
Comment (mering) [inline modules/fuzztest/20240813.0/patches/update_module_bazel_version.patch:10]: Same here.

Comment (udaya2899) [reply modules/fuzztest/20240813.0/patches/update_module_bazel_version.patch:10]: Argh. I'll clean up and generate a proper patch file again directly without manually copying the patch suggestion.

File: modules/fuzztest/20240813.0/presubmit.yml [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+  bazel:
+  - 7.x
+  - 6.x
+tasks:
+  verify_targets:
+    name: Verify build targets
+    platform: ${{ platform }}
+    bazel: ${{ bazel }}
+    build_flags:
+    - '--cxxopt=-std=c++17'
```
Comment (mering) [inline modules/fuzztest/20240813.0/presubmit.yml:17]: You might also need to check `--host-cxxopt`. As the current error is with Protobuf, you might want to copy the options from https://github.com/bazelbuild/bazel-central-registry/blob/ce1cc3491bf48880cf0a1ff7ed05779942a9b809/modules/protobuf/26.0/presubmit.yml#L13-L14.

**Comment by udaya2899** [comment]
@meteorcloudy is it actually logical/okay to skip having a presubmit for this module?  as the Source project itself has extensive bazel unit and e2e tests [here](https://github.com/google/fuzztest/blob/main/.github/workflows/bazel_test.yml) and [here](https://github.com/google/fuzztest/blob/main/.github/workflows/bazel_test_centipede.yml), the tests also would need [to run this shell script](https://github.com/google/fuzztest/blob/main/bazel/BUILD#L20).

**Comment by mering** [comment]
> @meteorcloudy is it actually logical/okay to skip having a presubmit for this module? as the Source project itself has extensive bazel unit and e2e tests [here](https://github.com/google/fuzztest/blob/main/.github/workflows/bazel_test.yml) and [here](https://github.com/google/fuzztest/blob/main/.github/workflows/bazel_test_centipede.yml), the tests also would need [to run this shell script](https://github.com/google/fuzztest/blob/main/bazel/BUILD#L20).

The test is using the same setup as any module depending on this. So if the test fails, it is very likely that this module will not be functional when depended upon by something else.

**Comment by udaya2899** [comment]
Ack. Thanks for the clarification @mering and @meteorcloudy. I'll try to add the fuzztest.bazelrc file and see if that makes a difference in the presubmits.

**Comment by bazel-io** [comment]
Hello @bazelbuild/bcr-maintainers, modules without existing maintainers (fuzztest) have been updated in this PR. Please review the changes.

**Comment by bazel-io** [comment]
Hello @fmeum, @linzhp, @tyler-french, modules you maintain (gazelle) have been updated in this PR. Please review the changes.

**Comment by udaya2899** [comment]
> Hello @fmeum, @linzhp, @tyler-french, modules you maintain (gazelle) have been updated in this PR. Please review the changes.

Please ignore, typo while playing aroud with presubmit fiels

**Comment by udaya2899** [comment]
@lszekeres finally! I got both of them working. Let's merge this PR, and I can upstream both patches in the CL in fuzztest and create a new version

**Comment by udaya2899** [comment]
@meteorcloudy the PR is ready for review now again.

File: modules/fuzztest/20240813.0/MODULE.bazel [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+    repo_name = "com_googlesource_code_re2",
+)
+
+bazel_dep(
+    name = "bazel_skylib",
+    version = "1.7.1",
+)
+
+bazel_dep(
+    name = "boringssl",
```
Comment (lszekeres) [inline modules/fuzztest/20240813.0/MODULE.bazel:25]: Note that this is not necessary from https://github.com/google/fuzztest/commit/b7257a64c701fc30265eb91179bf9310549bf440

**Review by lszekeres — APPROVED** [review]

**Review by fmeum — APPROVED** [review]

**Comment by dieortin** [comment]
@udaya2899 I'm getting the same error but with `com_google_googletest`, and I'm pretty sure it shouldn't be a `dev_dependency` either

**Comment by udaya2899** [comment]
@dieortin, sorry for the silence earlier. I sent a fix internally and hopefully fixing it at source directly instead of the patch.

**Pull request merged**